### PR TITLE
smtp_forward: add domain_selector

### LINKED ISSUE #2346: smtp_forward: support relaying based on MAIL FROM domain
The current plugin supports relaying based on looking the recipient domains. On the Cloudron, we allow setting a relay per domain. Any mail originating from the domain is sent via the configured relay.

### PR DESCRIPTION
**Pull request by gramakri**
this config variable can be set to rcpt_to or mail_from to decide how the domain configuration is chosen

Fixes #2346

Changes proposed in this pull request:
- add domain_selector config variable that chooses domain configuration based on mail_from or rcpt_to domain

Checklist:
- [ X ] docs updated
- [ X ] tests updated
- [ X ] [Changes](https://github.com/haraka/Haraka/blob/master/Changes.md) updated

### CODE REVIEW

File: plugins/queue/smtp_forward.js [hunk 1]
```diff
@@ -52,8 +52,15 @@ exports.get_config = function (connection) {
     const plugin = this;
 
     if (!connection.transaction) return plugin.cfg.main;
-    if (!connection.transaction.rcpt_to[0]) return plugin.cfg.main;
-    const dom = connection.transaction.rcpt_to[0].host;
+
+    var dom;
```
Comment (KingNoosh) [inline plugins/queue/smtp_forward.js]: No `var`, `let` or `const` only please.

File: plugins/queue/smtp_forward.js [hunk 2]
```diff
@@ -52,8 +52,15 @@ exports.get_config = function (connection) {
...
+    if (plugin.cfg.main.domain_selector === 'mail_from') {
+        if (!connection.transaction.mail_from) return plugin.cfg.main;
+        dom = connection.transaction.mail_from.host;
+    } else {
```
Comment (KingNoosh) [inline plugins/queue/smtp_forward.js]: No hugging elses, stroustrup brace style please

File: plugins/queue/smtp_forward.js (hunk 1 above)
Comment (gramakri) [reply plugins/queue/smtp_forward.js]: Fixed

File: plugins/queue/smtp_forward.js (hunk 2 above)
Comment (gramakri) [reply plugins/queue/smtp_forward.js]: Fixed this as well.

**Pull request merged**